### DISCUSSION TITLE
Can't Access Table name with custom Table

### DISCUSSION THREAD

**Discussion by JerContact** [simolus3/drift#2149] [Q&A]
I created this in a separate file:

```
CREATE VIRTUAL TABLE testTable USING FTS5 (
  Filename,
  FilePath,
  Description,
  Library,
  tokenize="porter unicode61"
);
```

Then, I build the whole .dart file and generated the generated file, but when I try to access the table it gives me the error (when doing a select(testTable).get():

The argument type 'Type' can't be assigned to the parameter type 'ResultSetImplementation<HasResultSet, dynamic>'.

On this call (bottom line):

```
final myIsolate = await createDriftIsolate();
final myDatabaseConnection = await myIsolate.connect();
final myDatabase = ManualDatabase.connect(myDatabaseConnection);
myDatabase.select(testTable).get();
```

I'm confused why i can't find this table somewhere in the generated drift classes....

**Comment by simolus3** [comment]
Did you include that file in the database (add `include: {'your_external_file.drift'}` to the `@DriftDatabase` annotation)? Are there any errors when you run the build?

**Reply by JerContact** [reply]
I'm getting no errors.  I have it all setup and it works fine if I make my own custom sql queries, but just trying to select a table through drift syntax doesn't work:

```
Future<DriftIsolate> createDriftIsolate() async {
  final dir = await getApplicationDocumentsDirectory();
  final path = p.join(dir.path, "ConstructDB.pkdb");
  final receivePort = ReceivePort();

  await Isolate.spawn(
    _startBackground,
    _IsolateStartRequest(receivePort.sendPort, path),
  );

  return await receivePort.first as DriftIsolate;
}

void _startBackground(_IsolateStartRequest request) {
  final executor = NativeDatabase(File(request.targetPath));

  final driftIsolate =
      DriftIsolate.inCurrent(() => DatabaseConnection(executor));

  request.sendDriftIsolate.send(driftIsolate);
}

class _IsolateStartRequest {
  final SendPort sendDriftIsolate;
  final String targetPath;

  _IsolateStartRequest(this.sendDriftIsolate, this.targetPath);
}

@DriftDatabase(include: {'tables.drift'})
class ManualDatabase extends _$ManualDatabase {
  ManualDatabase.connect(DatabaseConnection connection)
      : super.connect(connection);

  @override
  int get schemaVersion => 2;
}
```

If I create my own sql command in my tables.drift file, works fine:

```
getAll: SELECT * FROM testTable;
```

Then, later call it

```
myDatabase.getAll().get();
```

But trying to use the drift way, it gives an error.  I can see that table as a type, but the error pops up.  Am I supposed to re-make that table in the .dart file again or something so drift can see it properly?

This call:


`myDatabase.select(testTable).get();`

**Reply by JerContact** [reply]
The real question is more "Do I need to do somethings special to get the table of a custom table in a custom tables.drift file"?  B/c it doesn't seem that I can just directly reference it in the drift "select()".

**Reply by simolus3** [reply]
Did you mark `fts5` as an [available extension](https://drift.simonbinder.eu/docs/advanced-features/builder_options/#available-extensions) for the builder? That not being there might explain the table not getting generated, but the build should also print a warning in that case.

> Am I supposed to re-make that table in the .dart file again or something so drift can see it properly?

No, the point of drift files is that you just need the `CREATE (VIRTUAL) TABLE` statements. Well, at least that's the idea if they work :D

> B/c it doesn't seem that I can just directly reference it in the drift "select()".

Ideally it should work like this.

**Reply by JerContact** [reply]
I see the table being constructed in the .g.dart file, it has:

`class testTable extends Table with ...`

And has all the fields associated.  But when I use it in:

`select(testTable)`

It shows the error.  

The argument type 'Type' can't be assigned to the parameter type 'ResultSetImplementation<HasResultSet, dynamic>'.

**Reply by JerContact** [reply]
Yep, i have fts5 in there, it works totally fine if I use custom drift queries, but when i start to try and use all the functionality I just can't get it to work on a simple "select()" function, I must be missing something

**Reply by JerContact** [reply]
Got it working, not quite sure what happened.  Rewrote the database.dart file from scratch, deleted the .g.dart file, build it all again, and also just created it the normal way without drift isolates.

But it works as expected now!
